src/context/node.js

```diff
@@ -218,6 +218,22 @@ export class ContextNode {
       setTimeout
     );
 
+    // Shadow root: track slot changes.
+    if (node.nodeType == FRAGMENT_NODE) {
+      node.addEventListener('slotchange', (e) => {
+        const slot = /** @type {!HTMLSlotElement} */ (e.target);
+        // Rediscover newly assigned nodes.
```

Comment:
Nit: s/Rediscover/Discover (they've never been discovered before right?)

Reply:
Could have been discovered. Even likely. Shadow DOM only can distribute the nodes that were there before. So when a slot is added/changed the node's state changes. For instance, if we start with:

```
<div id="host">
  #shadow-root
    <slot name="slot1">
  <div id="child1"></div>
```

Then `child1` is essentially undisplayed and it's only connected to the `host`. But as soon as we change a slot assignment on it:

```
<div id="child1" slot="slot1"></div>
```

It gets connected to the shadow root via `slot1` and it potentially becomes displayed.

Reply:
A bit more context. At this stage we can't even discover new nodes that were not discovered before. But if a new node is discovered later (for instance if it's upgraded as a custom element later), then it will be discovered for the first time and will find the slot/shadow root automatically since they are now ready and expecting new nodes.